src/TLuaInterpreter.cpp

```diff
@@ -10754,6 +10755,55 @@ bool TLuaInterpreter::compileAndExecuteScript(const QString& code)
     }
 }
 
+// reformats given Lua code. In case of any issues, returns the original code as-is
+// issues could be invalid Lua code or the formatter code bugging out
+QString TLuaInterpreter::formatLuaCode(const QString &code)
+{
+    if (code.isEmpty()) {
+        return code;
+    }
+    lua_State* L = pIndenterState;
+    if (!L) {
+        qDebug() << "LUA CRITICAL ERROR: no suitable Lua execution unit found.";
+        return code;
+    }
+    
+    if (!validLuaCode(code)) {
+        return code;
+    }
+
+    QString escapedCode = code;
+    // escape backslashes so we can pass \n to the function
+    escapedCode.replace(QLatin1String("\\"), QLatin1String("\\\\"), Qt::CaseInsensitive);
```

Comment:
The `Qt::CaseInsensitive` is redundant and can be left to use the default I guess in this and the one two lines down?

As it happens I think there is an overload to `QString::replace(...)` which can take a `QChar:::LineFeed` {a `QChar::SpecialCharacter`} for those cases where you want exactly a LineFeed and wish to avoid confusion with a combination of back slashes and lower case `N`... :grinning:

Reply:
Fixed.

Not sure about using `QChar:::LineFeed`, I'll keep the consistency with the other replaces. `\n` is super common in MUDs to see whereas `QChar:::LineFeed` isn't...